prestashop1.7/controllers/admin/AdminTawktoController.php

```diff
@@ -221,8 +227,50 @@ public function ajaxProcessRemoveWidget()
      *
      * @return void
      */
-    public function ajaxProcessSetVisibility()
+    public function ajaxProcessSetOptions()
+    {
+        $key = TawkTo::TAWKTO_WIDGET_OPTS;
+        $jsonOpts = [];
+
+        try {
+            // Process selected options
+            $jsonOpts = $this->processSetOptions(Tools::getValue('options'));
+        } catch (Exception $e) {
+            die(json_encode(['success' => false, 'message' => $e->getMessage()]));
```

Comment:
We may be bleeding internals here in `e -> e.getMessage()` 
1) Log the error
2) Sub-class errors we want to propagate
3) Respond with error message only if handling sub-class, send generic failure message otherwise